core/pkg/launch/config_files.go

```diff
@@ -0,0 +1,64 @@
+package launch
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/wandb/wandb/core/internal/data_types"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+// Represents a config file parameter for a job.
+//
+// The relpath is the path to the config file relative to the files directory.
+// The includePaths and excludePaths are used to filter down the run config
+// before it is converted to a schema and saved as job inputs.
+type configFileParameter struct {
+	relpath      string
+	includePaths []ConfigPath
+	excludePaths []ConfigPath
+}
+
+func newFileInputFromRequest(
+	request *service.JobInputRequest,
+) (*configFileParameter, error) {
+	source := request.GetInputSource().GetSource()
+	file, ok := source.(*service.JobInputSource_File)
+	if !ok {
+		return nil, fmt.Errorf("jobBuilder: invalid source type for file input")
```

Comment:
This function should take a `*service.JobInputSource_File` parameter instead, since the caller already checks the type. The caller can use a type-switch:

```go
switch source := request.GetInputSource().GetSource().(type) {
case *service.JobInputSource_File:
	... := newFileInputFromRequest(source)
}
```

Reply:
great point

Reply:
this will simplify this function a lot!

Reply:
Yeah, good call!